conans/paths/package_layouts/package_user_layout.py

```diff
@@ -0,0 +1,47 @@
+# coding=utf-8
+
+import os
+
+from conans.paths import CONANFILE
+from conans.util.files import load
+from conans.model.ref import ConanFileReference
+from conans.model.ref import PackageReference
+
+CONAN_PACKAGE_LAYOUT_FILE = '.conan_package_layout'
+
+
+class PackageEditableLayout(object):
+
+    def __init__(self, linked_package_file, conan_ref):
+        assert isinstance(conan_ref, ConanFileReference)
+        self._conan_ref = conan_ref
+        self._base_folder = os.path.normpath(load(linked_package_file))
+
+    def conan(self):
+        """ Returns the base folder for this package reference """
+        return self._base_folder
+
+    def conanfile(self):
+        """ Path to the conanfile. We can agree that an editable package
+            needs to be a Conan package
+        """
+        return os.path.join(self.conan(), CONANFILE)
+
+    def installed_as_editable(self):
+        return True
+
+    def editable_package_layout_file(self):
+        return os.path.join(self.conan(), CONAN_PACKAGE_LAYOUT_FILE)
+
+    def package(self, package_reference):
+        assert isinstance(package_reference, PackageReference)
+        assert package_reference.conan == self._conan_ref
+        raise RuntimeError("Cannot retrieve this path for an editable package")
+        return self.conan()
+        return os.path.join(self.conan(), "package_reference")
```

Comment:
¿? asserts and raises and returns?

Reply:
We are working on it. What to do here? What to do in the `package_metadata` method? You are the one that should answer at least the second question.